lib/Rest/GeneralResponse.h

```diff
@@ -77,6 +78,10 @@ class GeneralResponse {
     _contentType = ContentType::CUSTOM;
   }
 
+
+  void setAllowCompression(bool allowed) { _allowCompression = allowed; }
```

Comment:
why don't we store the preferred response encoding type (e.g. deflate, unset, whatever) instead of a simple boolean?